src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -52,45 +52,16 @@ public void testDoGet() throws IOException, ServletException {
     Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
   }
 
-  @Test
-  public void testDoPost_BadUsername() throws IOException, ServletException {
-    Mockito.when(mockRequest.getParameter("username")).thenReturn("bad !@#$% username");
-
-    loginServlet.doPost(mockRequest, mockResponse);
-
-    Mockito.verify(mockRequest)
-        .setAttribute("error", "Please enter only letters, numbers, and spaces.");
-    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
-  }
-
-  @Test
-  public void testDoPost_NewUser() throws IOException, ServletException {
```

Comment:
I think one way to think about this is - could the LoginServlet still have to deal with a "new user" as an input? If yes, we should cover the case with a unit test. For example, lets say I tried to log in with a made up username (without registering). How should the LoginServlet respond?

Reply:
LoginServlet should respond with error! I will add that test into the tester file